### PR TITLE
docs: clarify tx-wide default max units explainer

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

unclear description of how tx-wide compute budget will calculate the default maximum units

#### Summary of Changes

attempt to clarify it

### CODE REVIEW

**Comment by jackcmay** [comment]
I updated docs in the backports while fixing the merge errors but this looks great too!

**Review by jackcmay — APPROVED** [review]

**Review by jstarry — APPROVED** [review]
Ah nice, this is much clearer, thanks!

File: docs/src/developing/programming-model/runtime.md [hunk 1]
```diff
@@ -113,10 +113,11 @@ Budget](#compute-budget).
 
 With a transaction-wide compute budget the `max_units` cap is applied to the
 entire transaction rather than to each instruction within the transaction. The
-default number of maximum units will calculated per instruction which means the
-sum of the compute units used by each instruction in the transaction must not
-exceed that value. The default number of maximum units allowed matches existing
-values to avoid breaking existing client behavior.
+transaction-wide default maximum number of units will be calculated as the product
+of the instruction count and the existing per-instruction maximum units. This means
```
Comment (jstarry) [inline docs/src/developing/programming-model/runtime.md:117]: Can you also mention that the default max is clamped at 1.4M compute units?

Comment (t-nelson) [reply docs/src/developing/programming-model/runtime.md:117]: Considered referencing the current constant values, but decided I didn't want to be responsible for keeping them up to date

Comment (jstarry) [reply docs/src/developing/programming-model/runtime.md:117]: Can we say something like "at the time of writing, this caps at 1.4M but it's subject to change" as a cop out? I think it's useful

Comment (t-nelson) [reply docs/src/developing/programming-model/runtime.md:117]: I see no precedent for citing other constant values in this file. Feel free to add them if you feel strongly!

**Pull request merged**